### PR TITLE
Add fetchGoalsFromPush method to read goals from subscription

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

File: lib/api-helper/goal/fetchGoalsOnCommit.ts [hunk 1]
```diff
@@ -35,6 +35,14 @@ import {
 import { goalKeyString } from "./sdmGoal";
 import { goalCorrespondsToSdmGoal } from "./storeGoals";
 
+export function fetchGoalsFromPush(sdmGoal: SdmGoalEvent): SdmGoalEvent[] {
+    const goals = sumSdmGoalEvents((sdmGoal.push as any).goals.filter(g => g.goalSetId === sdmGoal.goalSetId));
```
Comment (ddgenome) [inline lib/api-helper/goal/fetchGoalsOnCommit.ts]: Is `sdmGoal.push` not typed?

File: test/api-helper/goal/fetchGoalsOnCommit.test.ts [hunk 2]
```diff
@@ -0,0 +1,2043 @@
...
+      "approvalRequired": false,
+      "sha": "ac70626b35fc4873112794544d64aad0a2ae5269",
+      "approval": null,
+      "data": null
+    }, {
+      "externalKey": "sdm/atomist/0-code/build#machine.ts:77",
+      "description": "Build successful",
+      "goalSetId": "c2e95b8a-61a5-4474-95de-7d918d0b5478",
+      "preApproval": null,
+      "preCondit
```
Comment (ddgenome) [inline test/api-helper/goal/fetchGoalsOnCommit.test.ts]: That's some push.

Comment (ddgenome) [inline test/api-helper/goal/fetchGoalsOnCommit.test.ts]:
```suggestion
            assert(goals.every(g => g.push));
```

File: test/api-helper/goal/mock.test.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import assert = require("power-assert");
+import * as assert from "power-assert";
```
Comment (ddgenome) [inline test/api-helper/goal/mock.test.ts]: Aaawww yeah

File: lib/api-helper/goal/fetchGoalsOnCommit.ts (hunk 1 above)
Comment (cdupuis) [reply lib/api-helper/goal/fetchGoalsOnCommit.ts]: It is typed to a different type. Internally we are using subscription types but the API is using public `SdmGoalEvent`.

File: test/api-helper/goal/fetchGoalsOnCommit.test.ts (hunk 2 above)
Comment (cdupuis) [reply test/api-helper/goal/fetchGoalsOnCommit.test.ts]: That makes no sense.

**Pull request merged**